Added temperature map for ice planets and stuff

SlightlyIntelligentMonkey opened the pull request:
Added in a shader and code for generating a temperature map, which then determines where ice will be drawn (just pure white).  More complicated behavior could be added for better looking ice and a silly bug I'm unable to fix could be fixed (the shader uniform for the threshold for when a pixel is ice seems to be unable to be set properly).
Also made the roughness of land and water shader uniforms for more control, don't know how useful that'll be.

Commit:
Added temperature map for ice planets and stuff

crocodele commented on src/js/views/Planet.js:
```diff
@@ -305,15 +325,15 @@ class Planet {
   randomizeUqm() {
     this.randomize(1);
     this.uqmPlanetSeedChoice = this.seedString;
-
+    
     if (this.uqmPlanetSeedChoiceControl != null) {
       this.uqmPlanetSeedChoiceControl.updateDisplay();
     }
   }
 
   pickPlanetType() {
     let typeString = this.uqmPlanetType;
-    let type = this.uqmPlanetTable.findPlanetTypeByName(this.uqmPlanetType);
+    let type = this.UQM_PLANETTABLE.find(m => m.type == this.uqmPlanetType);
```
`this.UQM_PLANETTABLE` has been removed in https://github.com/pistolshrimpgames/ProceduralPlanet/pull/6, so this change should probably be reverted.

crocodele commented on src/js/views/Planet.js:
```diff
@@ -378,6 +398,15 @@ class Planet {
     }
   }
 
+  getUqmPlanetTypes() {
+    let table = this.UQM_PLANETTABLE;
+    let types = [];
+    for (let i = 0; i < table.length; i++) {
+      types.push(table[i].type);
+    }
+    return types;
+  }
```
This has also been removed, in favor of `UqmPlanetTable.getAllTypeNames()`.

Commit:
Extracted all of the rng factors for generation into a single class that's passed around.
Also added a .json to specify settings for different types of planet generation.

Commit:
Merge branch 'master' of https://github.com/pistolshrimpgames/ProceduralPlanet

Commit:
Finish merging in jewel worlds
